lib/parsers/asm-parser.ts

```diff
@@ -105,7 +105,7 @@ export class AsmParser extends AsmRegex implements IAsmParser {
         // Opcode expression here matches LLVM-style opcodes of the form `%blah = opcode`
         this.hasOpcodeRe = /^\s*(%[$.A-Z_a-z][\w$.]*\s*=\s*)?[A-Za-z]/;
         this.instructionRe = /^\s*[A-Za-z]+/;
-        this.identifierFindRe = /[$.@A-Z_a-z]\w*/g;
+        this.identifierFindRe = /([$.@A-Z_a-z]\w*)(?:@\w+)*/g;
```

Comment:
Is it important for the second group to be non-capturing? (I get that probably not; it's just all the other regexes capture-by-default here, and this sticks out as "different"). I'm not against it, but any time there's a dozen lines of regex and one clause is phrased a bit differently, it makes me think "is this important"

Reply:
The RE is used here https://github.com/compiler-explorer/compiler-explorer/pull/7522/files/630bf75948db66d0690b3d3cbd9c650474157214#diff-54e91a09afce28be680dd7812e6a572ebe10c206864de83b495515a3cb48b895L359-R359. If the second group was capturing, the replacer function would have to ignore it.

Reply:
Oh! Thank you! I didn't spot that 😊 - I obviously didn't look too hard either. This is now merged, and I'll deploy later today